### ISSUE TITLE
5.15.2 ?

### ISSUE THREAD

**Issue by nerohmot** [conda-forge/pyqt-feedstock#98]
As 5.15.2 is the last in the 5-series, it will be around for a while, the [PyPI package is there](https://pypi.org/project/PyQt5/#history), but this feed-stock is not updated yet ...

Someone has an ETA on this ?

@ccordoba12 : I know you are very busy with preparations for the all-new Spyder 5, but wouldn't it be a good foundation for Spyder 5 too ? I am thinking here for sure in the direction of the QScreenCast thingy, there we had some trouble with the 5.12.3 that had already a fix in 5.14.x I think, and as we gear up to re-do QScreenCast using GStreamer under the hood ...

Nice Bonus would be if pyqt==5.15.2 would also cover aarch64 (read: RPI & Co) :stuck_out_tongue_winking_eye:

**Comment by ccordoba12** [comment]
Hey @nerohmot, thanks for reporting. Unfortunately. to have PyQt 5.15 we first need to compile Qt 5.15, which is a huge undertaking. Some work was already done by @mingwandroid on that front, but I don't how advanced it is.

**Comment by mingwandroid** [comment]
It's causing a fire in my back burner Carlos!

**Comment by ccordoba12** [comment]
Haha, I'll try to give you a hand once we release Spyder 5 in April.

**Comment by nerohmot** [comment]
Oooh, I see, simple me was thinking that if there is a PyPI package, converting it to a conda-forge package would be easier ... (for someone more knowledgeable than me). When #84 was opened (I closed it in favor of this one) there simply was no PyPI pacage and thus the work had to be "manual" ...
Anyway to be continued :wink:

**Comment by mingwandroid** [comment]
Repackaging binaries from PyPI does not generally work well within the conda ecosystem. Binary compatibility is not guaranteed and we no longer depend on our own shared libraries or compiler runtimes.

Qt is just a huge amount of effort and every update gets worse as Qt gets bigger and bigger. With Qt 6 they've changed the build system from qmake to CMake and Qt 6 isn't quite ready yet (and the application ecosystem around it is far from ready). Still I have a hope that Qt 6 will be easier to build than Qt 5 is though I'm probably fooling myself somewhat! The other thing about it is that it makes the effort to update Qt 5 one that will not be useful for Qt 6.

I'd love to get back to updating Qt to the latest version. Those Windows builds are looking pretty lonely up on my anaconda.org channel!

**Comment by cbrnr** [comment]
Also, they've released 5.15.3 yesterday...

**Comment by hoechenberger** [comment]
Unfortunately, the current version of `pyqt` is also a huge burden for users running macOS 11, as Qt guis simply won't open … unless they set the environment variable `QT_MAC_WANTS_LAYER` to `"1"`.

**Comment by j9ac9k** [comment]
Right now Qt 5.12 is the oldest supported Qt version for pyqtgraph, we haven't established a timeline for making Qt 5.15 the minimum supported version; but I don't think we would do it without a package being available on `conda-forge`.   Don't mean to add to the noise here, just trying to relay that pyqtgraph will keep our conda users in mind.

**Comment by conda-forger** [comment]
It's been a year since the last update to this ticket.  Qt 5.15.2 is in conda-forge (as qt-main) but there are not corresponding PyQt bindings so it's not usable from Python apps.  I would very much like to be able to use Qt 5.15, and am willing to get involved and help work on this.

**Comment by BastianZim** [comment]
Note, that I'm not a maintainer of PyQt but I believe that the update of PyQt is currently underway. This was blocked by QT but since that was recently finished, we should see it here as well, including support for M1 etc.
Due to the size and complexity of it not all builds are being done in the feedstock so even if nothing is happening here, someone is probably working on it.

**Comment by cbrnr** [comment]
It would be great to get Qt6 (PySide6/PyQt6) into conda-forge. You might want to check out the current status here: https://github.com/conda-forge/qt-main-feedstock/pull/14

**Comment by ccordoba12** [comment]
@conda-forger, updating Qt/PyQt to 5.15 was a funded effort that took @andfoy nine months of really work to accomplish, which is really close to be finished (@BastianZim did a pretty good status update).

@cbrnr, yes, it'd be great but that really depends on the community because it won't be funded (at least not in the short term).

**Comment by cbrnr** [comment]
CC @hoechenberger

**Comment by h-vetinari** [comment]
This issue can be closed since https://github.com/conda-forge/pyqt-feedstock/pull/115 has been merged a few weeks ago. 🥳

**Comment by ccordoba12** [comment]
Right, closing then.
